Added a feedforward term for velocity commands in PID loop

davetcoleman opened the pull request:
In reference to @adolfo-rt 's comments last year:
https://github.com/ros-controls/ros_controllers/pull/64#discussion_r7727093
and
https://github.com/ros-controls/ros_controllers/commit/533ff43be53ea4996ffa9b2e079f889857e97483#joint_trajectory_controller-include-joint_trajectory_controller-hardware_interface_adapter-h-P14

Unfortuantly these changes were never made and when the PR was split up and refactored it was forgotten. 

Running this on Baxter, I saw huge improvements.

davetcoleman committed:
Added a feedforward term for velocity commands in PID loop

davetcoleman commented:
friendly ping?

adolfo-rt commented:
I'd be interested in knowing how well this patch behaves in the face of large disturbances. @davetcoleman, have you tried holding Baxter's arms so the tracking error grows significantly, and then let go?, are you satisfied with the behavior in such circumstances?.

If there's someone out there also using the velocity backend of this controller, would you mind giving this one a test?. I'm concerned that people already using it might have to change their existing PID gains specification if this patch gets merged.

davetcoleman commented:
I have not tried that, will test next time I'm using hardware.

davetcoleman commented:
I am still using this fix on Baxter and our new robot and its been over a year. Not having a feedforward term is really detrimental to the performance in my experience. I think this should be merged into Jade

bmagyar commented:
To put my two cents in, if this patch is merged I would also strongly consider adding this to indigo, since that is the LTS.

davetcoleman commented:
Perhaps now is the right time to merge this into the jade release?

adolfo-rt commented:
My only reservation here is that the PR might change behavior. Have you tried both alternatives (old and new) with the same set of gains, on hardware?.

davetcoleman commented:
Yes, it does change behavior. But that is why we release it into Indigo. Its a bug that I introduced when I implemented it a couple years ago, and I fixed it over a year ago, and still it remains. 

Currently, however, I am not using any velocity controllers so its hard for me to test it since I'm now using ros_control for other robots.
